adal/src/main/java/com/microsoft/aad/adal/AcquireTokenRequest.java

```diff
@@ -294,13 +297,16 @@ private AuthenticationResult acquireTokenSilentFlow(final AuthenticationRequest
     }
 
     /**
-     * Try acquire token silent locally.
-     */
+    * Try acquire token silent locally.
+    */
     private AuthenticationResult tryAcquireTokenSilentLocally(final AuthenticationRequest authenticationRequest)
             throws AuthenticationException {
         Logger.v(TAG, "Try to silently get token from local cache.");
         final AcquireTokenSilentHandler acquireTokenSilentHandler = new AcquireTokenSilentHandler(mContext,
                 authenticationRequest, mTokenCacheAccessor);
+        if(mAuthContext.getExtendedLifetimeEnabled()) {
+            acquireTokenSilentHandler.setOutageModeIsOn(true);
+        }
```

Comment:
Do you mean we only need a setter only to set the flag into true and never set it back to false? Or you mean to change the name of the method into setOutageMode?  @addev-ashish